tests/nn/data_parallel/test_sharded_ddp.py

```diff
@@ -109,10 +111,15 @@ def closure():
             same_params = device == torch.device("cpu") or grad_accumulation
             check_same_model_params(same_params=same_params)
 
-    check(broadcast_buffers=False)
-    check(broadcast_buffers=True)
-    check(broadcast_buffers=False, grad_accumulation=True)
-    check(broadcast_buffers=True, grad_accumulation=True)
+    for broadcast_buffers in [False, True]:
```

Comment:
I see several places here that you loop over what appears to be different test cases. Can they use "parameterized" annotator to be different tests? loop over the cases have several downsides:

- test can timeout
- test failure not clear which case failed
- not able to run tests in parallel

Thanks for adding this extensive tests though! The test is getting a bit too big. Is there a way to perhaps split the test files into multiple files?

Reply:
>> The test is getting a bit too big. Is there a way to perhaps split the test files into multiple files?
Well, it's always possible but then one would need to check two files instead of one ? I tried to refactor a little to save some lines, possible to do more, but all these tests are for the same class so it kind of make sense that they're in the same file, to me at least ? We've a ton of boilerplate code with pytest / multiprocess, it's clearly not really meant for that, but since pytest it is there's not that much I can do on that front